### PR TITLE
feat: add initial cli

### PR DESCRIPTION
**Pull request by keyvankhademi**

### CODE REVIEW

File: staking/cli/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+anchor-lang = {workspace = true}
+anchor-spl = {workspace = true}
+clap = {version ="3.2.22", features = ["derive"]}
+solana-client = "1.18.0"
+shellexpand = "3.1.0"
+pythnet-sdk = {version = "2.3.0", features = ["test-utils"]}
+bytemuck = {version = "1.4.0", features = ["derive", "min_const_generics"]}
+byteorder = "1.4.3"
+wormhole-vaas-serde =  "0.1.0"
+wormhole-solana-vaas = {version="0.3.0-alpha.1", features = ["anchor", "encoded-vaa", "mainnet"]}
```
Comment (guibescos) [inline staking/cli/Cargo.toml:21]: I think this dep is unused

File: staking/cli/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+serde_wormhole = "0.1.0"
+serde_json = "1.0.125"
```
Comment (guibescos) [inline staking/cli/Cargo.toml:23]: unused

File: staking/cli/Cargo.toml [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+solana-cli-output = "1.18.16"
```
Comment (guibescos) [inline staking/cli/Cargo.toml:24]: unused

File: staking/cli/Cargo.toml [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+spl-governance = { version = "4.0.0", features = ["no-entrypoint"] }
```
Comment (guibescos) [inline staking/cli/Cargo.toml:25]: unused

File: staking/cli/Cargo.toml [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+integrity-pool = {path = "../programs/integrity-pool"}
+publisher-caps = {path = "../programs/publisher-caps"}
+integration-tests = {path = "../integration-tests"}
+anchor-lang = {workspace = true}
+anchor-spl = {workspace = true}
+clap = {version ="3.2.22", features = ["derive"]}
+solana-client = "1.18.0"
+shellexpand = "3.1.0"
+pythnet-sdk = {version = "2.3.0", features = ["test-utils"]}
+bytemuck = {version = "1.4.0", features = ["derive", "min_const_generics"]}
```
Comment (guibescos) [inline staking/cli/Cargo.toml:18]: unused

File: staking/cli/src/cli.rs [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+    },
+    Advance {
+        #[clap(long, help = "Url to the hermes to fetch publisher caps")]
+        hermes_url: String,
+
+        #[clap(long, default_value = "3u8hJUVTA4jH1wYAyUur7FFZVQ8H635K3tSHHF4ssjQ5")]
+        wormhole: Pubkey,
+    },
+    InitializePoolRewardCustody {},
+    UpdatePythTokenMint {
```
Comment (guibescos) [inline staking/cli/src/cli.rs:66]: as said in the other pr I don't feel good keeping this instruction around

File: staking/cli/src/instructions.rs [hunk 7]
```diff
@@ -0,0 +1,472 @@
...
+    }
+    .to_account_metas(None);
+
+    let verify_encoded_vaa_instruction = Instruction {
+        program_id: wormhole,
+        accounts:   verify_encoded_vaa_accounts,
+        data:       wormhole_core_bridge_solana::instruction::VerifyEncodedVaaV1 {}.data(),
+    };
+
+    process_transaction(
```
Comment (guibescos) [inline staking/cli/src/instructions.rs:260]: note: on mainnet, the vaa is bigger so we'll need to do this in multiple transactions like in `write_publisher_caps`, but we can fix that once we need it

File: staking/cli/src/instructions.rs [hunk 8]
```diff
@@ -0,0 +1,472 @@
...
+        .decode(encoded_message)
+        .unwrap();
+
+    let (vaa, merkle_proofs) = deserialize_accumulator_update_data(message);
+
+
+    let encoded_vaa = process_write_encoded_vaa(rpc_client, vaa.as_slice(), wormhole, payer);
+
+
+    let publisher_caps = Keypair::new();
```
Comment (guibescos) [inline staking/cli/src/instructions.rs:440]: `init_publisher_caps` could return the pubkey for `publisher_caps`

File: staking/cli/src/instructions.rs [hunk 9]
```diff
@@ -0,0 +1,472 @@
...
+
+    let publisher_caps_message_bytes =
+        Vec::<u8>::from(merkle_proofs.first().unwrap().message.clone());
+
+
+    for i in (0..publisher_caps_message_bytes.len()).step_by(1000) {
+        let chunk =
+            &publisher_caps_message_bytes[i..min(i + 1000, publisher_caps_message_bytes.len())];
+
+        write_publisher_caps(rpc_client, payer, publisher_caps.pubkey(), i, chunk);
```
Comment (guibescos) [inline staking/cli/src/instructions.rs:454]: feels like init_puiblisher_caps + write_publisher_caps and verify_publisher_caps should all be in 1 function

**Review by guibescos — APPROVED** [review]

File: staking/cli/src/instructions.rs (hunk 9 above)
Comment (keyvankhademi) [reply staking/cli/src/instructions.rs:430]: they are all used in a function `fetch_publisher_caps_and_advance`

File: staking/cli/src/instructions.rs (hunk 8 above)
Comment (keyvankhademi) [reply staking/cli/src/instructions.rs]: done.

File: staking/cli/src/instructions.rs (hunk 7 above)
Comment (keyvankhademi) [reply staking/cli/src/instructions.rs:266]: sounds good.

File: staking/cli/src/cli.rs (hunk 6 above)
Comment (keyvankhademi) [reply staking/cli/src/cli.rs]: removed

**Pull request merged**